### PR TITLE
Rework of track fit and strategy

### PR DESCRIPTION
**Pull request by dhynds**
BEGINRELEASENOTES
- Rewrote track fit in SZ, proper phi treatment 
- Updated track strategy including extension of hits throughout the detector
- Utility tool added for debugging

ENDRELEASENOTES

### CODE REVIEW

**Comment by dhynds** [comment]
This seems to work well for single muons, picking up hits throughout the whole detector. There still seems to be an issue with unrelated hits being picked up for high pt tracks in "busier" events, currently being looked into.

File: src/ConformalTracking.cc [hunk 1]
```diff
@@ -322,6 +327,7 @@ void ConformalTracking::processEvent(LCEvent* evt) {
       kdClusterMap[kdhit] = hit;
       conformalHits[hit]  = kdhit;
       tempClusters.push_back(kdhit);
+      m_absZ->Fill(kdhit->getZ());
```
Comment (andresailer) [inline src/ConformalTracking.cc]: This crashes if debugPlots=off

**Comment by dhynds** [comment]
I have added encapsulation of the debug tool so that this runs fine without the flag set. The track fit changes recover performance for high pt tracks

**Comment by andresailer** [comment]
Thanks!
Is this OK to be merged now or do you still want to fix something else?

**Pull request merged**